module/__init__.py

```diff
@@ -689,9 +698,26 @@ def hoist_event(self, e):
         # this bit set. We don't actually care where the event came from, so we
         # just throw this away. Maybe we could expose this, if anyone actually
         # cares about it.
-        event = self._event_offsets[e.response_type & 0x7f]
+        response_type = e.response_type & 0x7f
 
         buf = CffiUnpacker(e)
+        event = None
+
+        # If the response is a GeGenericEvent then we can try to hoist this to relevant
+        # extension event.
+        if response_type == 35:
+            # Extract the extension opcode and event
+            extension, _, event_type, _ = buf.copy().unpack("xB2xIH22xI")
```

Comment:
I think you can do this as `buf.unpack(..., increment=False)` and save yourself a copy.